### PR TITLE
avoid creating `Map` in `dependentKeysDidChange`

### PR DESCRIPTION
**Pull request by bekzod**

### CODE REVIEW

File: packages/ember-metal/lib/property_events.js [hunk 1]
```diff
@@ -116,22 +116,23 @@ function notifyPropertyChange(obj, keyName, _meta) {
...
-    seen = DID_SEEN = new Map();
+  if (IS_TOP_SEEN_MAP) {
+    IS_TOP_SEEN_MAP = false;
   }
 
   iterDeps(notifyPropertyChange, obj, depKey, seen, meta);
 
-  if (top) {
-    DID_SEEN = null;
+  if (IS_TOP_SEEN_MAP) {
```
Comment (krisselden) [inline packages/ember-metal/lib/property_events.js]: Won't this always be false? did you mean to set the local `top = IS_TOP_SEEN_MAP` and use top in the conditionals?

Do we have a test that covers this?

**Review by krisselden — CHANGES_REQUESTED** [review]

File: packages/ember-metal/lib/property_events.js (hunk 1 above)
Comment (bekzod) [reply packages/ember-metal/lib/property_events.js]: thanks for catch, fixed, not sure how to test this, though it is doing same thing as before

**Comment by krisselden** [comment]
@rwjblue we definitely have lost test coverage if the seen map clearing doesn't matter

**Comment by krisselden** [comment]
I didn't mean to close it, I hit the wrong button

**Comment by bekzod** [comment]
yeah happens all the time to me ;P

**Review by krisselden — APPROVED** [review]

**Pull request merged**